src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -84,9 +91,12 @@
   private final String name;
   private final FirebaseOptions options;
   private final TokenRefresher tokenRefresher;
+  private final ThreadManager threadManager;
 
   private final AtomicBoolean deleted = new AtomicBoolean();
   private final Map<String, FirebaseService> services = new HashMap<>();
+  private final AtomicReference<ListeningScheduledExecutorService> executorReference =
```

Comment:
Since this is only ever accessed under lock, does it need to be an AtomicReference?